### ISSUE TITLE
Pull out larger datasets into their own packages

### ISSUE THREAD

**Issue by mine-cetinkaya-rundel** [OpenIntroStat/openintro#21]
...and make it a dependency of OpenIntro

**Comment by mine-cetinkaya-rundel** [comment]
Based on CRAN decision will need to rethink the size. A suggestion was to take out the functions from the package. These don't take a whole lot of space, but taking them out would make the pakcage a data-only package, which might be desirable to fall under the potential CRAN excemption for data-only packages that are not updated too regularly.

Here is an outline of next steps that will allow us to not ask for exemptions going forward as we add small-ish datasets:

1. Move large datasets into their own packages to reduce the size of openintro < 5MB so that there is some room for growth in this package in the near future.
1. Submit those packages to CRAN first and then submit openintro and have openintro install all packages with `install.packages(openintro)`, and load all of them with `library(openintro)` so for the student experience `library(openintro)` is sufficient. (Similar to `library(tidyverse)` experience.)
1. Going forward, aim to leave some room in the openintro package for growth (adding small datasets), but when using a LARGE dataset (let's arbitrarily say >500K, but we can revisit the threshold that) consider putting it in its own package that will be installed and loaded with openintro.
1. In the next openintro release, pull out functions that are used for making book figures (functions that students will likely not use, but instructors/developers forking and building on openintro resources might use) into their own package called ____ (I don't have a name for this yet). Doing this now would require a bit more thought as they're used in some of the examples, and I'd rather not add that to the scope now, but I can tackle it later. I'm also least sure of whether this is a required step, so we can think on it a bit more.

@DavidDiez see below for how I'm thinking of proceeding and let me know if you have any feedback. Also, note that I'd especially appreciate feedback on the last item re: county and state datasets.

## What's currently here?

```r
library(fs)
library(tidyverse)
```

Top 20 datasets, by size are

``` r
dir_info(path = "data/") %>% 
  arrange(desc(size)) %>% 
  slice_head(n = 20) %>% 
  select(path, size)
#>                          path    size
#> 1    data/county_complete.rda 808.75K
#> 2         data/usairports.rda 474.38K
#> 3    data/county_w_sm_ban.rda 472.46K
#> 4              data/birds.rda 467.05K
#> 5        data/seattlepets.rda    465K
#> 6         data/nycflights.rda 460.79K
#> 7                data/hfi.rda 458.76K
#> 8  data/loans_full_schema.rda 450.91K
#> 9              data/run17.rda 362.98K
#> 10               data/ipo.rda 345.88K
#> 11          data/run10_09.rda 336.37K
#> 12   data/sp500_1950_2018.rda 286.59K
#> 13   data/london_boroughs.rda 255.84K
#> 14             data/run10.rda 253.66K
#> 15             data/yrbss.rda 113.58K
#> 16              data/ames.rda  107.3K
#> 17            data/county.rda  92.74K
#> 18             data/email.rda  52.07K
#> 19          data/ucla_f18.rda   48.6K
#> 20            data/resume.rda  47.62K
```

## What should go where?

1. `usairports` is used in a single exercise in OS4 but is otherwise an interesting dataset of all airports in the US and some information about them. **Proposal:** Move it into a package called `usairports` or just `airports` (and maybe add other countries later). Both names are available on CRAN as of 2020-06-13. I think I like `airports` better for the potential to grow later.

2. Create a new package called `cherryblossom`, move all Cherry Blossom related datasets there (all but first one listed below). This way we have an opportunity to grow this going forward as well. Name is available on CRAN as of 2020-06-13 (and what a potential for a pretty hex!).

```r
dir_info(path = "data/") %>% 
  filter(str_detect(path, "run")) %>% 
  select(path, size)
#>                       path    size
#> 1 data/get_it_dunn_run.rda  14.38K
#> 2           data/run10.rda 253.66K
#> 3        data/run10_09.rda 336.37K
#> 4       data/run10samp.rda   1.43K
#> 5           data/run17.rda 362.98K
```

3. Both options below allow for growth. Option B allows for more growth, but then again, chances are state data can only get so large so maybe it doesn't matter.

*Option A:* Move US county datasets into their own package called `uscounties`, name available on CRAN as of 2020-06-13.

```r
dir_info(path = "data/") %>% 
  filter(str_detect(path, "county")) %>% 
  select(path, size)
#>                       path    size
#> 1          data/county.rda  92.74K
#> 2 data/county_complete.rda 808.75K
#> 3 data/county_w_sm_ban.rda 472.46K
```

*Option B:* Move US county and state related datasets into their own package called `usdata`, name available on CRAN as of 2020-06-13.

```r
dir_info(path = "data/") %>% 
  filter(str_detect(path, "county|state")) %>% 
  select(path, size)
#>                       path    size
#> 1          data/county.rda  92.74K
#> 2 data/county_complete.rda 808.75K
#> 3 data/county_w_sm_ban.rda 472.46K
#> 4     data/state_stats.rda   4.55K
```

**Comment by DavidDiez** [comment]
Thank you for all this work in prepping the package and figuring out plans! I have some questions and thoughts below, but generally the plan sounds good.

1. Github structure questions: Are you planning to create one repo per package, or will all the packages live in one repo? Or some other split?
1. Are the custom functions causing problems being in the `openintro` package? If not, what's the motivation for moving? You have *much* more experience with students using the package than me, so curious if students are wondering about the functions or get confused by the examples. Enumerating the options, at least those in my head: (a) Leave these as-is in `openintro`. (b) Migrate them to their own package. (c) Remove their usage from the Examples sections of data sets so that folks aren't inclined to use them unless they're digging around in greater detail. Given that you're bringing this topic up, I'm assuming (a) isn't okay. :)
1. Related to 500kB threshold: Suppose we leave a data set `X` in the package that is modestly large (e.g. 200kB) but not so large that it's currently a problem. When `openintro` later becomes too big again, we may need to lower our threshold and move `X` to its own package. Making sure I understand the plan in that case: we'll create that new package and submit it to CRAN, then we'll remove it from the `openintro` package and have the other package be a dependency. Do I have that right?
1. Any idea how big is too big for a data package on CRAN? Asking because I wonder if any packages we create will eventually grow to be notably bigger over time, e.g. a hypothetical `usdata` package.
1. The `airports` package idea sounds great to me.
1. Not all the data sets noted for `cherryblossom` are from the Cherry Blossom Run. We may also want to include more running data later. More generic names might be `run`, `exercise`, etc. That said, `cherryblossom` sounds more fun, so if the aim is to have the package name be fun and more unique vs being being extremely explicit, then I'm on board with `cherryblossom`.
1. For the county and/or state data, either option is philosophically okay with me. I'll be curious about a possible technical hangup around the `usdata` for if that package might get too bloated (per a few questions above). For example, what if there's a big US data dump that we want to include and qualifies? Would it go in this package, or would we make it its own package that is then a dependency of `usdata`? I feel like I'm going a bit down a rabbit hole when this is a much smaller question, so feel free to simply choose and move on. :)

Again, generally sounds good. You've got a better handle on the details and know the nuances, so whatever you end up deciding will be good with me.

**Comment by mine-cetinkaya-rundel** [comment]
1. 1 repo / packge so we can do `install_github("OpenIntroStat/PKGNAME")` easily. I found the old structure confusing, and I'd prefer to keep it standard with how many others who develop R packages on GitHub keep things.

2. My inclination is to go with (a) for now, and then evaluate in a future release of the package. Some of it is probably confusing for some of the audience (e.g. student learning R with OpenIntro labs never encounters `hisPlot()`, they encounter either `ggplot()` or `hist()`. But there are a few nice custom functions that I think might be useful for educators, e.g. `lmPlot()` is handy for slides where screen real-estate is limited and you just want to show that residual plot along with the scatterplot. So, in a future release what I want to do is to turn some of the more extensive examples into a vignette that uses these custom functions (for instructors and for students who have learned a bit of R already) and keep the examples simpler (for complete newcomers). Once I do that, I might go with (b) if it seems like space is an issue. I feel like it's ok to punt on this since these functions aren't taking too much space, hence not super relevant to current block.

3. Yes, that's the plan. But that's kinda why I chose 500kB as opposed to 1MB, to minimize that as much as possible.

4. 5MB is the "limit", and exceptions are made but not guaranteed.

5. ✅ 

6. 4/5 are, but agreed. I kinda like the `cherryblossom` data (maybe because I like those flowers) so I'm inclined to keep it going with that name.

7. Ok, I'll make a last-minute decision on this one. We probably don't have to think too far into the future for the other packages, but I wanted to have a clear plan for the `openintro` package and develop thoughts around the others as we use them.

Thanks for the feedback. I'm going to work on getting these thru as soon as possible.

if I go with (b) in a future release, I'm planning on also doing (c) and then moving some of the more extensive examples that use these custom functions into a vignette for OpenIntro.
